src/app/jscaip/MGPNode.ts

```diff
@@ -2,42 +2,38 @@ import { Move } from './Move';
 import { SCORE } from './SCORE';
 import { Rules } from './Rules';
 import { MGPMap } from '../utils/MGPMap';
-import { LegalityStatus } from './LegalityStatus';
 import { assert, display, Utils } from 'src/app/utils/utils';
 import { NodeUnheritance } from './NodeUnheritance';
 import { Minimax } from './Minimax';
 import { MGPSet } from '../utils/MGPSet';
 import { MGPOptional } from '../utils/MGPOptional';
 import { Player } from './Player';
-import { AbstractGameState } from './GameState';
+import { GameState } from './GameState';
+import { MGPFallible } from '../utils/MGPFallible';
 
 export class MGPNodeStats {
     public static createdNodes: number = 0;
     public static minimaxTime: number = 0;
 }
 
-export class MGPNode<R extends Rules<M, S, L>,
+export class MGPNode<R extends Rules<M, S, L>, // TODO FOR REVIEW: why not remove R and use Rules<M, S, L> instead of R?
                      M extends Move,
-                     S extends AbstractGameState,
-                     L extends LegalityStatus = LegalityStatus,
+                     S extends GameState,
+                     L = void,
                      U extends NodeUnheritance = NodeUnheritance> {
     // TODO: calculate a board - value by the information of the mother.boardValue + this.move to ease the calculation
-    // TODO: choose ONE commenting langage, for fuck's sake.
```

Comment:
is it done :p ?

Reply:
mostly oui, je suis encore tombé à un endroit sur des commentaires en FR mais sinon tout ce qu'on peut trouver via grep est traduit ! Y'a encore pas mal de francisation dans le code mais ça, ça se fera petit à petit